src/Tooltip.tsx

```diff
@@ -0,0 +1,111 @@
+import React, { CSSProperties, forwardRef, memo } from "react";
+import type { Equals } from "tsafe";
+import { assert } from "tsafe/assert";
+import { useAnalyticsId } from "./tools/useAnalyticsId";
+import { fr } from "./fr";
+import { cx } from "./tools/cx";
+
+export type UseTooltipParams = UseTooltipParams.Common &
+    (UseTooltipParams.WithTriggerOnClick | UseTooltipParams.WithTriggerOnHover);
+
+export namespace UseTooltipParams {
+    export type Common = {
+        id?: string;
+        className?: string;
+        style?: CSSProperties;
+    };
+
+    export type WithTriggerOnClick = {
+        trigger: "click";
+        buttonLabel?: string;
+    };
+
+    export type WithTriggerOnHover = {
+        trigger?: "hover";
+        buttonLabel?: never;
+    };
+}
+
+type UseTooltipReturn = {
+    tooltipOwnerProps: {
+        "aria-describedby": string;
+        className?: string;
+    };
+    TooltipOwnerButtonComponent?: JSX.Element;
```

Comment:
```suggestion
    button?: JSX.Element;
```

This is not a component; it's a node. Therefore, it shouldn't be capitalized.

Additionally, I think the API could benefit from a redesign. Whether the button is undefined or not should be inferred from the parameters. If a button exists, then tooltipOwnerProps has no reason to be prsent on the return type. Also, it's somewhat confusing from the user's perspective to have to both mount a node and a component. Ideally, they should just have to use a single TooltipButton with a `label` and a `tooltip` prop or something similar.

When designing an API, it's crucial to consider the user's perspective. What makes the most sense for them. What design would constiture a better (DX). The implementation details are secondary; the primary focus should be on fulfilling the API design.

Reply:
Hi @garronej, thank you for your review :)
I updated the PR and made a new suggestion for the Tooltip Component.